components/ws-manager-bridge/src/workspace-instance-controller.ts

```diff
@@ -0,0 +1,308 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { TraceContext } from "@gitpod/gitpod-protocol/lib/util/tracing";
+import { GetWorkspacesRequest } from "@gitpod/ws-manager/lib";
+import { DisposableCollection, RunningWorkspaceInfo, WorkspaceInstance } from "@gitpod/gitpod-protocol";
+import { inject, injectable } from "inversify";
+import { Configuration } from "./config";
+import { log, LogContext } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { PrometheusMetricsExporter } from "./prometheus-metrics-exporter";
+import { WorkspaceDB } from "@gitpod/gitpod-db/lib/workspace-db";
+import { DBWithTracing, TracedUserDB, TracedWorkspaceDB } from "@gitpod/gitpod-db/lib/traced-db";
+import { UserDB } from "@gitpod/gitpod-db/lib/user-db";
+import { MessageBusIntegration } from "./messagebus-integration";
+import { PrebuildUpdater } from "./prebuild-updater";
+import { IAnalyticsWriter } from "@gitpod/gitpod-protocol/lib/analytics";
+import { ClientProvider } from "./wsman-subscriber";
+import { repeat } from "@gitpod/gitpod-protocol/lib/util/repeat";
+
+export const WorkspaceInstanceController = Symbol("WorkspaceInstanceController");
+
+export interface WorkspaceInstanceController {
+    start(
+        installation: string,
+        clientProvider: ClientProvider,
+        controllerIntervalSeconds: number,
+        controllerMaxDisconnectSeconds: number,
+    ): void;
+
+    controlNotStoppedAppClusterManagedInstanceTimeouts(
```

Comment:
This is a mouthful. How about 
```suggestion
    handleZombies(
```
or something similar. We can introduce a noun which corresponds to this Java like expression to improve mind association

Reply:
I'd like to keep the explicit name, as it avoids confusion, and fits into the pattern of a similarly named function in the same class.